File: 345. Reverse Vowels of a String.cpp
class Solution {
public:
    string reverseVowels(string s) {
        string vowel = "AEIOUaeiou";
        int l = 0, r = s.size()-1;
        while (l <= r)
        {
            size_t foundl = vowel.find(s[l]), foundr = vowel.find(s[r]);
            if (foundl != string::npos && foundr != string::npos)
            {
                char temp = s[l];
                s[l] = s[r];
                s[r] = temp;
                l ++;
                r --;
            }
            else if (foundl == string::npos && foundr != string::npos)
                l ++;
            else if (foundl != string::npos && foundr == string::npos)
                r --;
            else
            {
                l ++;
                r --;
            }
        }
        return s;
    }
};

File: 345. Reverse Vowels of a String.py
class Solution(object):
    def reverseVowels(self, s):
        """
        :type s: str
        :rtype: str
        """
        vowels = ['A','E','I','O','U','a','e','i','o','u']
        l = 0
        r = len(s)-1
        s = list(s)
        while l <= r:
            if s[l] in vowels and s[r] in vowels:
                s[l], s[r] = s[r], s[l]
                l += 1
                r -= 1
            elif s[l] in vowels and s[r] not in vowels:
                r -= 1
            elif s[l] not in vowels and s[r] in vowels:
                l += 1
            else:
                l += 1
                r -= 1
        return ''.join(s)
